Moodlet/app/src/main/java/com/cmput3owo1/moodlet/fragments/RegisterFragment.java

```diff
@@ -0,0 +1,119 @@
+package com.cmput3owo1.moodlet.fragments;
+
+import android.content.Intent;
+import android.os.Bundle;
+
+import androidx.annotation.NonNull;
+import androidx.fragment.app.Fragment;
+import androidx.fragment.app.FragmentTransaction;
+
+import android.text.TextUtils;
+import android.view.LayoutInflater;
+import android.view.View;
+import android.view.ViewGroup;
+import android.widget.Button;
+import android.widget.EditText;
+import android.widget.TextView;
+import android.widget.Toast;
+
+import com.cmput3owo1.moodlet.R;
+import com.cmput3owo1.moodlet.activities.LoginActvity;
+import com.google.android.gms.tasks.OnCompleteListener;
+import com.google.android.gms.tasks.Task;
+import com.google.firebase.auth.AuthResult;
+import com.google.firebase.auth.FirebaseAuth;
+import com.google.firebase.firestore.CollectionReference;
+import com.google.firebase.firestore.FirebaseFirestore;
+
+import java.util.HashMap;
+
+public class RegisterFragment extends Fragment {
+
+    EditText fullname, username, email, password, confirmPassword;
+    TextView loginText;
+    Button registerButton;
+
+    FirebaseAuth auth;
+    FirebaseFirestore db;
+
+    public RegisterFragment() {
+        // Required empty public constructor
+    }
+
+
+    @Override
+    public View onCreateView(LayoutInflater inflater, ViewGroup container, Bundle savedInstanceState) {
+        // Inflate the layout for this fragment
+        View registerFragmentView = inflater.inflate(R.layout.fragment_register, container, false);
+
+        fullname = registerFragmentView.findViewById(R.id.full_name);
+        username = registerFragmentView.findViewById(R.id.username);
+        email = registerFragmentView.findViewById(R.id.email);
+        password = registerFragmentView.findViewById(R.id.password);
+        confirmPassword = registerFragmentView.findViewById(R.id.confirm_password);
+        registerButton = registerFragmentView.findViewById(R.id.btn_register);
+        loginText = registerFragmentView.findViewById(R.id.swap_to_login_text_view);
+
+        auth = FirebaseAuth.getInstance();
+        db = FirebaseFirestore.getInstance();
+
+        final CollectionReference collectionReference = db.collection("users");
+
+        loginText.setOnClickListener(new View.OnClickListener() {
```

Comment:
same thing with the click listeners in this file. Make a function called "onLoginTextClick(View v)" and in the XML set the onClick property to this function

Reply:
I'll do that in a separate PR